velox/vector/VariantToVector.cpp

```diff
@@ -77,4 +77,158 @@ ArrayVectorPtr variantArrayToVector(
       pool);
 }
 
+struct NestedTypeCounter {
```

Comment:
NIT: Can you add brief comments on what this structure is supposed to represent and how it will be used.

Reply:
add docstring.